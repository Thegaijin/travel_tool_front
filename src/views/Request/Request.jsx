import React, { PureComponent } from 'react';
import './Request.scss';
import RequestPanelHeader from '../../components/request-header/RequestPanelHeader';

class Request extends PureComponent{
    render(){
        return(
          <RequestPanelHeader />
        )
    }
}

export default Request;