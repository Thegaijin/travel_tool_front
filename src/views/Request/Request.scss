.RequestsPanelHeader {
    background-color: #F8F8F8;
    height: 201.9px;
    width: 1200px;
    padding: 0 100px 0 60px;

    .NewRequests {
        display: flex;
        justify-content: space-between;
        align-items: center;
        position: relative;
        top: 50px;
        height: 36px;
        margin-bottom: 46px;
    
        .requests {
            height: 23px;
            width: 89px;
            color: #4F4F4F;
            font-family: "DIN Pro";
            font-size: 18px;
            font-weight: 500;
            line-height: 23px;
        }
        .NewRequestButton{
            height: 36px;	
            width: 124px;	
            border: 1px solid #3359DB;	
            border-radius: 1px;	
            background-color: #FFFFFF;
        }
        .NewRequestButtonText{
            height: 18px;	
            width: 84px;	
            color: #3359DB;	
            font-family: "DIN Pro";	
            font-size: 14px;	
            font-weight: 500;	
            line-height: 18px;	
            text-align: center;
        }
    }

    .OpenRequests{
        display: flex;
        justify-content: space-between;
        align-items: center;
        position: relative;
        top: 50px;
        height: 36px;
        
        .OpenRequestsButtonSection{
            width: 348px;
            display: flex;       
            height: 36px;
            justify-content: space-between;
     
            .all-button {
             height: 36px;
             width: 60px;
             border: 1px solid #3359DB;
             border-radius: 1px;
             background-color: #3359DB;
                 .all{
                     height: 18px;	
                     width: 18px;	
                     color: #FFFFFF;	
                     font-family: "DIN Pro";	
                     font-size: 14px;	
                     font-weight: 500;	
                     line-height: 18px;	
                     text-align: right;
                 }
           }
           .open-request-button{
             display: flex;
             justify-content: space-around;
             align-items: center;  
             height: 36px;
             width: 145px;
             border: 1px solid #D4D4D4;
             border-radius: 1px;	
             background-color: #F8F8F8;
             .open-requests {
                 height: 18px;
                 width: 95px;
                 color: #999999;
                 font-family: "DIN Pro";
                 font-size: 14px;
                 font-weight: 500;
                 line-height: 14px;
                 text-align: right;
             }
             .oval-4 {	
                height: 15px;	
                width: 15px;	
                background-color: #3359DB;
                display: flex;
                align-items: center;
                justify-content: center;
                border-radius: 50%;
            }
             .open-request-number{
                 margin-bottom: 0px !important;
                 height: 15px;	
                 width: 7px;	
                 color: #FFFFFF;	
                 font-family: "DIN Pro";	
                 font-size: 12px;	
                 font-weight: 500;	
                 line-height: 15px;	
                 text-align: center;
             }
           }
           .past-request-button{
             height: 36px;	
             width: 123px;	
             border: 1px solid #D4D4D4;	
             border-radius: 1px;	
             background-color: #F8F8F8;
             .past-requests {
                 height: 18px;
                 width: 93px;
                 color: #999999;
                 font-family: "DIN Pro";
                 font-size: 14px;
                 font-weight: 500;
                 line-height: 14px;
             }
           }
         }

         .ItemsPerPageSection{
            height: 36px;	
            width: 174px;	
            border: 1px solid #D4D4D4;	
            border-radius: 1px;	
            display: flex;
            justify-content: space-around;
            align-items: center;
            .ItemsPerPageTextSection{
                width: 116px;
                height: 36px;
                background-color: #F8F8F8;
                border: 1px solid #D4D4D4;	
                border-radius: 1px;	
                display: flex;
                justify-content: space-around;
                align-items: center;
            }
            .ItemsPerPageText{
                height: 18px;	
                width: 96px;	
                color: #999999;	
                font-family: "DIN Pro";	
                font-size: 14px;	
                font-weight: 500;	
                line-height: 14px;	
                text-align: right;
            }
            .divider{
                box-sizing: border-box;	
                height: 36px;	
                width: 0px;	
                border: 1px solid #D4D4D4;
            }

            .ItemsPerPagePagination{
                height: 34px;	
                width: 55px;	
                border-radius: 0 2px 2px 0;	
                background-color: #FFFFFF;
                display: flex;
                justify-content: space-between;
                align-items: center;

                .PaginateCounter{
                    height: 20px;	
                    width: 16px;	
                    color: #3359DB;	
                    font-family: "DIN Pro";	
                    font-size: 15px;	
                    font-weight: 500;	
                    line-height: 20px;	
                    text-align: center;
                }
                .PaginatorDropdown{
                    height: 24px;	
                    width: 24px;	
                    background-color: #FFFFFF;
                }
            }
         }
    }

    

}

