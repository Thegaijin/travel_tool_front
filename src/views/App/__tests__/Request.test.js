import React from 'react';
import renderer from 'react-test-renderer';
import Request from '../../Request/Request';

it('should render request header correctly', () => {
    const tree = renderer.create(<Request />).toJSON();
    expect(tree).toMatchSnapshot();
});