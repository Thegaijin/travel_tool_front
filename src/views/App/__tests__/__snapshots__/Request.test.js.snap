// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render request header correctly 1`] = `
<div
  className="RequestsPanelHeader"
>
  <div
    className="NewRequests"
  >
    <div>
      <span
        className="requests"
      >
        REQUESTS
      </span>
    </div>
    <div>
      <button
        className="NewRequestButton"
        type="button"
      >
        <span
          className="NewRequestButtonText"
        >
          New Request
        </span>
      </button>
    </div>
  </div>
  <div
    className="OpenRequests"
  >
    <div
      className="OpenRequestsButtonSection"
    >
      <button
        className="all-button"
        type="button"
      >
        <span
          className="all"
        >
          All
        </span>
      </button>
      <button
        className="open-request-button"
        type="button"
      >
        <span
          className="open-requests"
        >
          Open Requests
        </span>
        <span
          className="oval-4"
        >
          <p
            className="open-request-number"
          >
            3
          </p>
        </span>
      </button>
      <button
        className="past-request-button"
        type="button"
      >
        <span
          className="past-requests"
        >
          Past Requests
        </span>
      </button>
    </div>
    <div
      className="ItemsPerPageSection"
    >
      <div
        className="ItemsPerPageTextSection"
      >
        <span
          className="ItemsPerPageText"
        >
          Items per page
        </span>
      </div>
      <div
        className="divider"
      />
      <div
        className="ItemsPerPagePagination"
      >
        <div>
          <span
            className="PaginateCounter"
          >
            10
          </span>
        </div>
        <div
          className="PaginatorDropdown"
        >
          <span>
            <img
              alt="Andela Logo"
              src="Icon.svg"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
